projects/eventonica/eventonica-part7-react.md

```diff
@@ -72,10 +70,145 @@ Here's an example of what `package.json` might look like now:
+    return (
+      <div id="all-users">
+        {eventonica.allUsers.map((user) => (
+          <li>
+            {user.userName}, ID: {user.userID}
+          </li>
+        ))}
+      </div>
+    );
+  }
+}
+```
+
+Try migrating your version of the users list and see if you can get it to display. Note you're still not using props yet, but we're trying to move in small increments to make our progress more manageable.
+
+### Use React props to pass data
+
+Instead of globally referencing `eventonica`, we want to make our app follow the React pattern of using props.
+
+```jsx
+// src/index.js - where <App> is created and mounted to the DOM
+import Eventonica from './models.js';
+
+const eventonica = new Eventonica();
+eventonica.addUser('Lisa', 12345);
+// ... any other starter data you want
+
+ReactDOM.render(
+  <React.StrictMode>
+    <App eventonica={eventonica} />
+  </React.StrictMode>,
+  document.getElementById('root')
+);
+
+// src/App.js
+class App extends React.Component {
+  render() {
+    const { eventonica } = this.props;
+    return (
+      <div>
+        <UsersList users={eventonica.allUsers} />
+      </div>
+    );
+  }
+}
+```
+
+Now you will have to modify `UsersList.js` to accept this prop. If you change the seed data above, CRA should live reload with the changes.
+
+### Event Handling
+
+In your Part 2 version, you attached event handlers to the DOM. With React, you will use props like `onClick` to achieve the same result. Again, this example code may not be exactly how your code works but hopefully you can see how to apply the idea to yours.
+
+#### JS DOM Example of Handling a User Delete Action
+
+```js
+document
+  .querySelector('#delete-user-action')
+  .addEventListener('click', (event) => {
+    event.preventDefault();
+    const idToDelete = parseInt(
+      document.querySelector('#delete-user-id').value
+    );
+    eventonica.deleteEvent(idToDelete);
+    displayEvents(); // refresh the DOM
+  });
+```
+
+In React we would setup the event handler like this:
+
+```jsx
+// src/UserDeleteForm.js
+class UserDeleteForm extends React.Component {
+  state = { idToDelete: null };
+
+  render() {
+    const onSubmit = (event) => {
+      event.preventDefault();
+      eventonica.deleteUser(this.state.idToDelete);
+    };
+    return (
+      <form onSubmit={onSubmit}>
+        <label>User ID:</label>
+        <input
+          id="delete-user-id"
+          type="number"
+          onChange={(e) => {
+            this.setState({ idToDelete: e.target.value });
+          }}
+          placeholder="1234"
+        />
+        <input id="delete-user-action" type="submit" value="Delete User" />
+      </form>
+    );
+  }
+}
+```
+
+### Re-rendering the UI in Response to Changes
+
+Now we want the UI to re-render when we delete the user. To do this we will have to call `this.setState` with updated data to ensure that React will re-render. Use the previous lessons you went through to figure out how to do this.
```

Comment:
Please review this new wording, @alodahl . The last wording was confusing and came from me having mixed results teaching immutable code patterns as the participants might see in a real codebase vs. modifying and objects in state and then just calling `this.setState` with the whole object again.

Reply:
One thing that's really important for React that you don't have to care about with jQuery is that the user ID list should be in the state of the parent component, meaning that the onClick will need to be a callback prop, and you probably want the callback function to live in the parent to make it clear.  If you think they've learned enough to accomplish this, maybe you can just leave a note reminding them of these things, but let them write the code themselves?